docs/examples/openiotsdk_examples.md

```diff
@@ -232,33 +253,143 @@ You can call the script directly yourself.
+1. Click 'Run and Debug' from the primary side menu or press (Ctrl+Shift+D)
+2. Select 'Debug Open IoT SDK example application' from the drop down list
+3. Click 'Start Debugging'(green triangle) or press (F5)
+4. Enter:
+=> <example name>
+=> (GDB target address)
+=> (network namespace)
+=> (network interface)
+=> <example name>
 ```
 
 For debugging remote targets (i.e. run in other network namespaces) you need to
-pass hostname/IP address of external GDB target that you want to connect to
-(_GDB target address_). In case of using the
+pass the hostname/IP address of the external GDB target that you want to connect
+to (_GDB target address_). In the case of using the
 [Open IoT SDK network environment](#networking-setup) the GDB server runs inside
-a namespace and has the same IP address as bridge interface.
+a namespace and has the same IP address as the bridge interface.
 
 ```
 ${MATTER_ROOT}/scripts/run_in_ns.sh <namespace_name> ifconfig <bridge_name>
 ```
 
-**NOTE**
+**NOTES**
 
 As you can see above, you will need to select the name of the example twice.
 This is because the debug task needs to launch the run task and currently VS
 code has no way of passing parameters between tasks.
+
+There are issues with debugging the lock-app example if you happen to be using
+"--network=host" in your docker container configuration and you are trying to
+debug while connected to a VPN. The easiest solution is just to come off the VPN
+while debugging.
+
+## Specific examples
+
+### Build and run the lock-app example using the CLI
+
+```
+${MATTER_ROOT}/scripts/examples/openiotsdk_example.sh -s -b psa -S lwip lock-app
+```
+
+```
+export TEST_NETWORK_NAME=OIStest
+```
+
+```
+sudo ${MATTER_ROOT}/scripts/examples/scripts/setup/openiotsdk/network_setup.sh -n $TEST_NETWORK_NAME down
+```
+
+```
+sudo ${MATTER_ROOT}/scripts/examples/scripts/setup/openiotsdk/network_setup.sh -n $TEST_NETWORK_NAME up
+```
+
+```
+${MATTER_ROOT}/scripts/examples/scripts/run_in_ns.sh ${TEST_NETWORK_NAME}ns
+scripts/examples/openiotsdk_example.sh -C test -n ${TEST_NETWORK_NAME}tap
+lock-app
+```
+
+### Build and run the lock-app example using tasks
+
+```
+Command Palette (F1), type: tasks <return>
+=> Build Open IoT SDK example
+=> Use debug mode (False)
+=> Crypto algorithm to use (psa)
+=> Socket API to use (iotsocket)
+=> Example application to use (lock-app)
+```
+
+In CLI:
+
+```
+sudo ${MATTER_ROOT}/scripts/examples/scripts/setup/openiotsdk/network_setup.sh -n $TEST_NETWORK_NAME down
+```
+
+```
+sudo ${MATTER_ROOT}/scripts/examples/scripts/setup/openiotsdk/network_setup.sh -n $TEST_NETWORK_NAME up
+```
+
+```
+Command Palette (F1), type: tasks <return>
+=> Run Open IoT SDK example
+=> Network namespace:  OIStestns
+=> Network interface name: OIStesttap
+=> Example application to use (lock-app)
+```
+
+The example output should be seen in the terminal window.
+
```

Comment:
There is a lot of output to the terminal , especially when building the example . What would you suggest we show for that? Just the couple of lines showing compilation successful?

Reply:
No, I mean to add a new section that shows command usage for `build and test` lock-app. We have `build and run`, `build and debug`. I think it will be worth adding the same for the testing step also.

Reply:
What is build and test then? Not sure I see something different ?